vmm/src/vm.rs

```diff
@@ -450,9 +452,24 @@ impl Vm {
                     .checked_add(KERNEL_64BIT_ENTRY_OFFSET)
                     .ok_or(Error::MemOverflow)?;
 
-                Ok(GuestAddress(load_addr))
+                Ok(EntryPoint {
+                    entry_addr: GuestAddress(load_addr),
+                    protocol: BootProtocol::LinuxBoot,
+                })
             }
             None => {
+                // Assume by default Linux boot protocol is used and not PVH
+                let mut entry_point_addr: GuestAddress = entry_addr.kernel_load;
+
+                let boot_prot = if cfg!(feature = "pvh_boot") && entry_addr.pvh_entry_addr.is_some()
```

Comment:
If the feature is turned on then this code correctly checks for bzImage/vmlinux vs Xen PVH. i.e. the same CH binary can load all the kernel types?

Reply:
> If the feature is turned on then this code correctly checks for bzImage/vmlinux vs Xen PVH. 

IIUC , the check in the code is more bzImage vs ELF boot (and then ELF boot has two flavors: 64bit Linux boot and PVH)
Turning on the feature just makes the VMM use the PVH boot protocol as its first choice when using ELF boot, but if the kernel does not support PVH boot (i.e. the KernelLoaderResult field called pvh_entry_addr is None) then it will boot using direct ELF boot as it does now.

IIRC, QEMU tries multiboot, PVH boot, and then direct ELF boot, in that order. I just didn't want to assume that PVH would be the preferred mode, so I put it behind a feature.

> the same CH binary can load all the kernel types?
Yes, if pvh_boot is turned on.

Reply:
Thanks for your explanation. I'm not sure we need the feature control if we have tested with all three kernel types. FYI the firmware will soon be coming as an ELF PVH binary so we will need to make sure that continues to work.